### ISSUE TITLE
Precip fraction in P3 incorrectly defined

### ISSUE THREAD

**Issue by AaronDonahue** [E3SM-Project/scream#146]
The precip fraction (i.e. rcldm) as defined in the `get_cloud_fraction` subroutine in P3 is not defined correctly and should be fixed.  When using `max_overlap` the cloud fraction doesn't consider the rain cloud fraction in the layer above, it just considers the liquid/ice cloud fraction.  This is not the correct implementation of the `max_overlap` approach.

A second issue is whether or not sedimentation should be based on cloud fraction of rain cloud fraction.  Currently the fall speed for each water species is dependent on the cloud fraction for that species.  This may be incorrect for two reasons, 1) sedimentation is a precip based process so shouldn't it use the precip fraction (rcldm)? 2) If we only use the liquid/ice cloud fraction to determine fallspeeds doesn't this mean that we will only get liquid or ice surface precip when we have fog (i.e. liquid cloud at the surface)?  Note we can still have rain at the surface since the precip cloud fraction takes into account the cloud fraction above the surface layer.

The latter question is why this is an issue since the first question is a simple fix.  But we should think a bit more about how we want to handle sedimentation in terms of sub-grid cloudiness.

**Comment by AaronDonahue** [comment]
PR #167 addresses the first issue stated above.

**Comment by PeterCaldwell** [comment]
I don't think ice sedimentation should use rain area fraction, for example - it should definitely use ice cloud fraction. 

I *do* think there's potentially a problem here in the sense that it looks like ice can't fall below cloud base because sedimentation always loops over a top to bottom which is calculated before we start substepping in time for CFL stability. See https://github.com/E3SM-Project/scream/blob/02f62bdbe09e626beb6a01734caeb6088f30cd31/components/eam/src/physics/cam/micro_p3.F90#L3939 .  A symptom of this should be that we don't have much snowpack in our model. I think the solution is to subtract 1 from k_qxbot every sedimentation sub-timestep.

I just confirmed that sedimentation of each species uses that species' cloud fraction. I don't think it makes sense

**Comment by PeterCaldwell** [comment]
@AaronDonahue - I think (or at least hope!) we dealt with this issue ages ago. True?

**Comment by AaronDonahue** [comment]
Yes. I believe that #167 solved it, at least the first part of the issue.  As for the second part, I think your comment addresses that.

**Comment by bartgol** [comment]
Is this still an issues? As we get close to v1, we might want to make sure high priority issues are tackled (though this one might not be a v1 issue, but rather a v1.1?).
